### ISSUE TITLE
Provide way to import projection trait

### ISSUE THREAD

**Issue by taiki-e** [taiki-e/pin-project#80]
The current project/project_into methods are trait-methods, so it needs to use glob import to use it from other modules.

We could probably provide the following way:

```rust
#[pin_project]
struct Foo<T> {
    #[pin]
    field: T,
}
mod bar {
    #[project]
    use super::Foo;
    // convert to
    use super::__FooProjectionTrait;
}
```

**Comment by taiki-e** [comment]
Blocked on https://github.com/taiki-e/pin-project/issues/21#issuecomment-530206201

**Comment by withoutboats** [comment]
Didn't search very hard but: why do you need the trait? I thought we made pin `#[fundamental]`?

**Comment by taiki-e** [comment]
Closing this in favor of #89.
